configs/components/_base-ruby-augeas.rb

```diff
@@ -48,6 +48,11 @@
     ruby = "#{host_ruby} -r#{settings[:datadir]}/doc/rbconfig-#{ruby_version}-orig.rb"
     pkg.environment "LDFLAGS", settings[:ldflags]
   end
+elsif platform.is_macos?
+  pkg.environment "PATH", "$(PATH):/opt/homebrew/bin"
+  pkg.environment 'CC', 'clang'
+  pkg.environment "LDFLAGS", settings[:ldflags]
+  ruby = File.join(ruby_bindir, 'ruby')
```

Comment:
I think something about this is breaking macOS 11/12 Intel. Maybe the `LDFLAGS` settting? Not sure why `augeas-devel` would suddenly be missing?

```
17:41:12 export PATH="/usr/bin:/bin:/usr/sbin:/sbin:/opt/homebrew/bin" && \
17:41:12 	export CONFIGURE_ARGS="--vendor" && \
17:41:12 	export PKG_CONFIG_PATH="/opt/puppetlabs/puppet/lib/pkgconfig:/usr/lib/pkgconfig" && \
17:41:12 	export CC="clang" && \
17:41:12 	export LDFLAGS="-L/opt/puppetlabs/puppet/lib" && \
17:41:12 	cd ruby-augeas-0.5.0 && \
17:41:12 	/usr/bin/patch --strip=2 --fuzz=0 --ignore-whitespace --no-backup-if-mismatch < ../ruby-augeas-0.5.0-patch_c_extension.patch && \
17:41:12 	/opt/puppetlabs/puppet/bin/ruby ext/augeas/extconf.rb && \
17:41:12 	/usr/bin/make -e -j3
17:41:12 patching file ext/augeas/_augeas.c
17:41:12 *** ext/augeas/extconf.rb failed ***
17:41:12 Could not create Makefile due to some reason, probably lack of necessary
17:41:12 libraries and/or headers.  Check the mkmf.log file for more details.  You may
17:41:12 need configuration options.
17:41:12 
17:41:12 Provided configuration options:
17:41:12 	--with-opt-dir
17:41:12 	--with-opt-include
17:41:12 	--without-opt-include=${opt-dir}/include
17:41:12 	--with-opt-lib
17:41:12 	--without-opt-lib=${opt-dir}/lib
17:41:12 	--with-make-prog
17:41:12 	--without-make-prog
17:41:12 	--srcdir=ext/augeas
17:41:12 	--curdir
17:41:12 	--ruby=/opt/puppetlabs/puppet/bin/$(RUBY_BASE_NAME)
17:41:12 	--with-augeas-dir
17:41:12 	--without-augeas-dir
17:41:12 	--with-augeas-include
17:41:12 	--without-augeas-include=${augeas-dir}/include
17:41:12 	--with-augeas-lib
17:41:12 	--without-augeas-lib=${augeas-dir}/lib
17:41:12 	--with-augeas-config
17:41:12 	--without-augeas-config
17:41:12 	--with-pkg-config
17:41:12 	--without-pkg-config
17:41:12 ext/augeas/extconf.rb:27:in `<main>': augeas-devel not installed (RuntimeError)
```